src/js/providers/tracks-mixin.ts

```diff
@@ -918,28 +919,25 @@ function _kindSupported(kind: string): boolean {
     return kind === 'subtitles' || kind === 'captions';
 }
 
-function getTextCueMetaEvent(cue: TrackCue): MetadataEvent | null {
+function getTextCueMetaEvent(cue: TrackCue): ProviderEvents['meta'] | null {
     let metadata;
     try {
         metadata = JSON.parse(cue.text);
     } catch (e) {
         return null;
     }
-    const event: MetadataEvent = {
+    const event: ProviderEvents['meta'] = {
+        metadataType: metadata.metadataType,
+        metadataTime: cue.startTime,
         metadata
     };
-    event.metadataTime = cue.startTime;
     if (metadata.programDateTime) {
         event.programDateTime = metadata.programDateTime;
     }
-    if (metadata.metadataType) {
-        event.metadataType = metadata.metadataType;
-        delete metadata.metadataType;
```

Comment:
is this delete no longer relevant?

Reply:
It was not relevant - `metadata` is an object we just parsed from JSON `cue.text` that is not what is being returned. and has no use outside of the scope of this function.